assembly/src/lib.rs

```diff
@@ -151,6 +222,15 @@ impl Assembler {
     // IMPORT PARSERS
     // --------------------------------------------------------------------------------------------
 
+    /// Add an instance of the provider to the internal list
+    pub fn add_module_provider<P>(&mut self, provider: P)
```

Comment:
minor nit: I'd probably put this into a separate section (as it is not an "import parser"). Maybe something like "state mutators" right after "public accessors".

Reply:
Agreed, its a bit off in the parsers section. Updated!